File: queries/src/main/java/bigframe/queries/Query.java
package bigframe.queries;

import java.util.HashSet;
import java.util.Set;

/**
 * The class provide basic information of a query.
 * 
 * @author andy
 *
 */
public abstract class Query {
	protected Set<String> dataTypes;
	protected String queryType;
	protected String description;
	protected String runningEngine;
	
	public Query() {
		dataTypes = new HashSet<String>();
			
	}
	
	public void setDataTypes(Set<String> dataTypes) {
		this.dataTypes = dataTypes;
	}
	
	public Set<String> getDataTypes() {
		return dataTypes;
	}
	
	public String getqueryType() {
		return queryType;
	}
	
	public void setqueryType(String queryType) {
		this.queryType = queryType;
	}
	
	public String getDescription() {
		return description;
	}
	
	public void setDescription(String des) {
		description = des;
	}
	
	public abstract void printDescription();
	
}

File: workflows/src/main/scala/bigframe/workflows/Query.scala
package bigframe.workflows


import scala.collection.mutable.Set

/**
 * The class provide basic information of a query.
 * 
 * @author andy
 *
 */
abstract class Query {
	protected var _dataTypes : Set[String] = Set()
	protected var _queryType = ""
	protected var _description = ""
	protected var _runningEngine = ""
	
	/**
	 * Getters
	 */
	def dataTypes = _dataTypes
	
	def queryType = _queryType
	
	def description = _description
	
	def runningEngine = _runningEngine
	
	/**
	 * Setters
	 */
	def dataTypes_= (value : Set[String]): Unit = _dataTypes = value
	
	def queryType_= (value : String): Unit = _queryType = value
	
	def description_= (value : String): Unit = _description = value
	
	def runningEngine_= (value : String): Unit = _runningEngine = value
	
	/**
	 * Print out the description of this query
	 */
	def printDescription(): Unit
	
}
